Forte.EpiServer.AzureSearch/Events/BlockSearchEventHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+using System.Linq;
+using EPiServer;
+using EPiServer.Core;
+using EPiServer.DataAbstraction;
+using Forte.EpiServer.AzureSearch.Extensions;
+using Forte.EpiServer.AzureSearch.Model;
+
+namespace Forte.EpiServer.AzureSearch.Events
+{
+    public class BlockSearchEventHandler<T> where T : ContentDocument
+    {
+        private readonly IContentSoftLinkRepository _linkRepository;
+        private readonly IContentRepository _contentRepository;
+        private readonly IContentDocumentBuilder<T> _contentDocumentBuilder;
+
+        public BlockSearchEventHandler(IContentSoftLinkRepository linkRepository, IContentRepository contentRepository, IContentDocumentBuilder<T> contentDocumentBuilder)
+        {
+            _linkRepository = linkRepository;
+            _contentRepository = contentRepository;
+            _contentDocumentBuilder = contentDocumentBuilder;
+        }
+
+        public IEnumerable<T> GetBlockContentDocuments(ContentReference contentLink)
+        {
+            var contentInAllLanguages = _contentRepository.GetAllLanguageVersions(contentLink);
+            var documents = contentInAllLanguages
+                .Select(_contentDocumentBuilder.Build)
+                .ToList();
+            return documents;
+        }
+        
+        public IEnumerable<ContentReference> GetBlockParentPages(ContentReference contentLink)
+        {
+            var parents = new List<ContentReference>();
+            
+            AddParentsFromContentReferences(contentLink, parents);
+            AddParentsFromXhtmlProperties(contentLink, parents);
+
+            return parents.DistinctBy(parent => parent.ID);
+        }
+        
+        private void AddParentsFromContentReferences(ContentReference contentLink, List<ContentReference> parents)
+        {
+            var referencesToContent = _contentRepository.GetReferencesToContent(contentLink, false);
```

Comment:
GetReferencesToContent method is marked as "Unsupported INTERNAL API!"

Reply:
I can't see it, both in IDE and on their docs.

https://world.episerver.com/csclasslibraries/cms/EPiServer.IContentRepository?version=11#EPiServer_IContentRepository_GetReferencesToContent_EPiServer_Core_ContentReference_System_Boolean_

Reply:
Could it change in epi 11.17? Because the project references 11.13

Reply:
I saw your answer to this posted in another question, so I will resolve this one.